pkg/keyidentifier/testdata/make.sh

```diff
@@ -3,97 +3,170 @@
 # This script uses several tools to create ssh keys. We attempt to be
 # as exhaustive as possible to create a wide range of things to test.
 #
-# This puts them into a somewhat ugly filesystem hierarchy. The intent
-# is to group things to make it simpler for tools to test
-# groups. The layout should straightforward.
-
-
+# each function also generates a json spec file that describes various
+# attributes of a key, plus the name of the key described. This spec file is
+# consumed by golang tests, and the output of the keyidentifier package is
+# compared against these expected values.
 
 set -e
 #set -x
 
 function rand {
-    dd if=/dev/random bs=1 count=16 2> /dev/null  | base64
+    dd if=/dev/random bs=1 count=16 2> /dev/null  | base64 | sed 's/[^0-9a-zA-Z]//g'
```

Comment:
```suggestion
   LC_ALL=C tr -dc 'A-Za-z0-9' </dev/urandom | head -c 16
```

For nitpicky reasons I'm not fond of this. the `count=16`, means you always have two trailing `=`, which sed just strips out anyhow. And you get inconsistent file lengths. The internet helped me with this code golf